cmake/DefaultTargetProperties.cmake

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(set_default_target_properties target_name)
+    #-------------------------------------------------------------------
+    # C++ standard
+
+    # Minimum C++ standard
+    target_compile_features(${target_name} PRIVATE cxx_std_14)
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Ensure the proposed compiler supports our minimum C++
+        # standard.
+        CXX_STANDARD_REQUIRED ON
+        # Disable compiler extensions. E.g. use -std=c++11  instead of
+        # -std=gnu++11.  Helps limit cross-platform issues.
+        CXX_EXTENSIONS OFF
+    )
+
+    #-------------------------------------------------------------------
+    # Compiler warnings
+
+    include(CompilerWarnings)
+    set_default_compiler_warnings(${target_name})
+
+
+    #-------------------------------------------------------------------
+    # Potentially enable sanitizers.
+    include(Sanitizers)
+    enable_sanitizers(project_options)
+
+
+    #-------------------------------------------------------------------
+    # Symbol visibility
+
+    # Hide symbols from this library by default.
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        C_VISIBILITY_PRESET hidden
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN YES
+    )
+
+    # Hide all symbols from external statically linked libraries.
+    if (CMAKE_CXX_COMPILER_ID MATCHES ".*Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_link_options(${target_name} PRIVATE -Wl,--exclude-libs,ALL)
+    endif ()
+
+    # Use C++11 ABI by default, but allow disabling.
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+        CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+        option(GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" ON)
+        if (GLIBCXX_USE_CXX11_ABI)
+            target_compile_definitions(${target_name} PRIVATE _GLIBCXX_USE_CXX11_ABI=1)
+        else ()
+            target_compile_definitions(${target_name} PRIVATE _GLIBCXX_USE_CXX11_ABI=0)
+        endif ()
+    endif ()
+
+    # TODO(DF): POSITION_INDEPENDENT_CODE required for Python module? See OCIO.
```

Comment:
If it's a shared library, fPIC is assumed. I think that you _need_ to have a shared library for Python bindings (don't quote me on that though).

In our thirdparty library builds, we also build static libraries with fPIC enabled, because you need that if you link the static library into a downstream shared library.

Reply:
Interesting point about linking as a static library to a downstream shared library.  I wonder if we should just always set it, then, given OpenAssetIO's positioning as an intermediary?

Reply:
Windows modules? I read the link, and that's odd, because Python C extension modules are dynamic libraries for all platforms, in all the distributions I've seen.

I see no harm in specifying fPIC to be the most flexible. (Although as mentioned, redundant if you only ever build shared libs.)

Reply:
Added an option to force PIC (enabled by default) in dbc19a9f